federatedscope/autotune/utils.py

```diff
@@ -171,7 +255,32 @@ def eval_in_fs(cfg, config, budget, client_cfgs=None):
                             server_class=get_server_cls(trial_cfg),
                             client_class=get_client_cls(trial_cfg),
                             config=trial_cfg.clone(),
-                            client_config=client_cfgs)
+                            client_configs=client_cfgs)
     results = Fed_runner.run()
     key1, key2 = trial_cfg.hpo.metric.split('.')
-    return results[key1][key2]
+    return results[key1][key2], results
+
+
+def config_bool2int(config):
+    # TODO: refactor bool/str to int
+    import copy
+    new_dict = copy.deepcopy(config)
+    for key, value in new_dict.items():
+        if isinstance(new_dict[key], bool):
+            new_dict[key] = int(value)
+    return new_dict
+
+
+def log2wandb(trial, config, results, trial_cfg):
+    try:
+        import wandb
+        key1, key2 = trial_cfg.hpo.metric.split('.')
+        log_res = {
+            'Trial_index': trial,
+            'Config': config_bool2int(config),
+            trial_cfg.hpo.metric: results[key1][key2],
+        }
+        wandb.log(log_res)
+    except ImportError:
```

Comment:
I didn't see any necessity to have a try-catch block from this exception handling step.